tests/unit/module_utils/test_s3.py

```diff
@@ -5,82 +5,425 @@
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import (absolute_import, division, print_function)
-
 __metaclass__ = type
 
-from ansible_collections.amazon.aws.tests.unit.compat.mock import MagicMock
+import pytest
+import random
+import string
+
 from ansible_collections.amazon.aws.plugins.module_utils import s3
-from ansible.module_utils.basic import AnsibleModule
+from unittest.mock import MagicMock, patch, call
 
-import pytest
+try:
+    import botocore
+except ImportError:
+    pass
+
+
+def generate_random_string(size, include_digits=True):
+    buffer = string.ascii_lowercase
+    if include_digits:
+        buffer += string.digits
+
+    return ''.join(random.choice(buffer) for i in range(size))
+
+
+@pytest.mark.parametrize("parts", range(10))
+@pytest.mark.parametrize("version", [True, False])
+def test_s3_head_objects(parts, version):
+
+    client = MagicMock()
+
+    s3bucket_name = "s3-bucket-%s" % (generate_random_string(8, False))
+    s3bucket_object = "s3-bucket-object-%s" % (generate_random_string(8, False))
+    versionId = None
+    if version:
+        versionId = random.randint(0, 1000)
+
+    total = 0
+    for head in s3.s3_head_objects(client, parts, s3bucket_name, s3bucket_object, versionId):
+        assert head == client.head_object.return_value
```

Comment:
You're comparing mocked output with mocked output. It would be better to initialize `client.head_object.return_value` before.

Reply:
This is something I got from the documentation, it avoid having return_value when it is not used anywhere else in the code